---
layout: archive
title: "Selected Publications"
permalink: /publications/
author_profile: true
---


The full list of publications is available on <u><a href="https://scholar.google.com/citations?user=h4_CmPoAAAAJ&hl=en">my Google Scholar profile</a>.</u>

\* authors contributed equally; # co-corresponding authors

## 2024

* Sheng Yang, and Xiang Zhou (2024). <a href="https://spatialtranscriptomicsanalysis.com">SRT-Server</a>: Powering the analysis of spatial transcriptomic data. Genome Medicine. 16: 18.
* Lu Liu, Ran Yan, Ping Guo, Jiadong Ji, Weiming Gong, Fuzhong Xue, Zhongshang Yuan, and Xiang Zhou (2024). Conditional transcriptome-wide association study for fine-mapping candidate causal genes. Nature Genetics. 56: 348–356.
* Boran Gao, and Xiang Zhou (2024). MESuSiE: Multi-ancestry fine-mapping for scalable and powerful discovery of shared and ancestry-specific putative causal variants in genome-wide association studies. Nature Genetics. 56: 170-179.

## 2023

* Chang Xu, Santhi Ganesh, and Xiang Zhou (2023). mtPGS: Leverage multiple correlated traits for accurate polygenic score construction. American Journal of Human Genetics. 110: 1673-1689.
* Lulu Shang, Wei Zhao, Yi Zhe Wang, Zheng Li, Jerome J. Choi, Minjung Kho, Thomas H. Mosley, Sharon L.R. Kardia, Jennifer A. Smith, and Xiang Zhou (2023). meQTL mapping in the GENOA study reveals genetic determinants of DNA methylation in African Americans. Nature Communications. 14: 2711.
* Jiaqiang Zhu\*, Lulu Shang\*, and Xiang Zhou (2023). SRTsim: Spatial pattern preserving simulations for spatially resolved transcriptomics. Genome Biology. 24: 39.

## 2022

* Lulu Shang, and Xiang Zhou (2022). Spatially aware dimension reduction for spatial transcriptomics. Nature Communications. 13: 7203.
* Zheng Li, and Xiang Zhou (2022). BASS: Multi-scale and multi-sample analysis enables accurate cell type clustering and spatial domain detection in spatial transcriptomic studies. Genome Biology. 23: 168.
* Peiyao Zhao, Jiaqiang Zhu, Ying Ma, and Xiang Zhou (2022). Modeling zero inflation is not necessary for spatial transcriptomics. Genome Biology. 23: 118.
* Fred Boehm, and Xiang Zhou (2022). Statistical methods for Mendelian randomization in genome-wide association studies: A review. Computational and Structural Biotechnology Journal. 20: 2338-2351.
* Ying Ma, and Xiang Zhou (2022). Spatially informed cell type deconvolution for spatial transcriptomics. Nature Biotechnology. 40: 1349–1359.
* Zheng Li, Wei Zhao, Lulu Shang, Thomas H Mosley, Sharon LR Kardia, Jennifer A Smith, and Xiang Zhou (2022). METRO: Multi-ancestry transcriptome-wide association studies for powerful gene-trait association detection. American Journal of Human Genetics. 109: 783-801.
* Sheng Yang, and Xiang Zhou (2022). <a href="https://pgs-server.com">PGS-Server</a>: Accuracy, robustness, and transferability of polygenic score methods for biobank scale studies. Briefings in Bioinformatics. 22: bbac039.
* Jialu Hu, Mengjie Chen, and Xiang Zhou (2022). Effective and scalable single-cell data alignment with non-linear canonical correlation analysis. Nucleic Acids Research. 50: e21.
* Zhongshang Yuan, Lu Liu, Ping Guo, Ran Yan, Fuzhong Xue, and Xiang Zhou (2022). Likelihood based Mendelian randomization analysis with automated instrument selection and horizontal pleiotropic modeling. Science Advances. 8: eabl5744.

## 2021

* Jiaqiang Zhu, Shiquan Sun, and Xiang Zhou (2021). SPARK-X: Non-parametric modeling enables scalable and robust detection of spatial expression patterns for large spatial transcriptomic studies. Genome Biology. 22: 184.
* Lu Wang, Boran Gao, Yue Fan, Fuzhong Xue, and Xiang Zhou (2021). Mendelian randomization under the omnigenic architecture. Briefings in Bioinformatics. 22: bbab322.
* Ying Ma, and Xiang Zhou (2021). Genetic prediction of complex traits with polygenic scores: A statistical review. Trends in Genetics. 37: 995-1011. 
* Ping Zeng, Zhonghe Shao, and Xiang Zhou (2021). Statistical methods for mediation analysis in the era of high-throughput genomics: Current successes and future challenges. Computational and Structural Biotechnology Journal. 19: 3209-3224.
* Huanhuan Zhu, and Xiang Zhou (2021). Transcriptome-wide association studies: A view from Mendelian randomization. Quantitative Biology. 9: 107–121
* Lu Liu, Ping Zeng, Fuzhong Xue, Zhongshang Yuan, and Xiang Zhou (2021). Multi-trait transcriptome-wide association studies with probabilistic Mendelian randomization. American Journal of Human Genetics. 108: 240-256.
* Ping Zeng, Jing Dai, Siyi Jin, and Xiang Zhou (2021). Aggregating multiple expression prediction models improves the power of transcriptome wide association studies. Human Molecular Genetics. 30: 939–951.
* Huanhuan Zhu\*, Lulu Shang\*, and Xiang Zhou (2021). A review of the statistical methods for identifying trait-relevant tissues and cell types. Frontiers in Genetics. 11: 587887.
* Yue Fan, Huanhuan Zhu, Yanyi Song, Qinke Peng, and Xiang Zhou (2021). Efficient and effective control of confounding in eQTL mapping studies through joint differential expression and Mendelian randomization analyses. Bioinformatics. 37: 296–302.
* Boran Gao, Can Yang, Jin Liu, and Xiang Zhou (2021). Accurate genetic and environmental covariance estimation with composite likelihood in genome-wide association studies. PLOS Genetics. e1009293.

## 2020

* Zhongshang Yuan, Huanhuan Zhu, Ping Zeng, Sheng Yang, Shiquan Sun, Can Yang, Jin Liu, and Xiang Zhou (2020). Testing and controlling for horizontal pleiotropy with the probabilistic Mendelian randomization in transcriptome-wide association studies. Nature Communications. 11: 3861.
* Tae Hyun Kim, Xiang Zhou#, and Mengjie Chen# (2020). Demystifying "drop-outs" in single-cell UMI data. Genome Biology. 21: 196.
* Huanhuan Zhu, and Xiang Zhou (2020). Statistical methods for SNP heritability estimation and partition: A review. Computational and Structural Biotechnology Journal. 18: 1557-1568.
* Sheng Yang, and Xiang Zhou (2020). Accurate and scalable construction of polygenic scores in large biobank data sets. American Journal of Human Genetics. 106: 679-693.
* Lulu Shang, Jennifer A. Smith, and Xiang Zhou (2020). Leveraging gene co-expression patterns to infer trait-relevant tissues in genome-wide association studies. PLOS Genetics. 16: e1008734.
* Lulu Shang\*, Jennifer A. Smith\*, Wei Zhao, Minjung Kho, Stephen T. Turner, Thomas H. Mosley, Sharon L.R. Kardia#, and Xiang Zhou# (2020). Genetic architecture of gene expression in European and African Americans: An eQTL mapping study in GENOA. American Journal of Human Genetics. 106: 496-512.
* Ying Ma\*, Shiquan Sun\*, Xuequn Shang, Evan T. Keller, Mengjie Chen and Xiang Zhou (2020). Integrative differential expression and gene set enrichment analysis using summary statistics for scRNA-Seq studies. Nature Communications. 11: 1585.
* Yanyi Song, Xiang Zhou#, Min Zhang, Wei Zhao, Yongmei Liu, Sharon Kardia, Ana Diez Roux, Belinda Needham, Jennifer A Smith and Bhramar Mukherjee# (2020). Bayesian shrinkage estimation of high dimensional causal mediation effects in omics studies. Biometrics. 76: 700-710.
* Shiquan Sun\*, Jiaqiang Zhu\* and Xiang Zhou (2020). Statistical analysis of spatial expression pattern for spatially resolved transcriptomic studies. Nature Methods. 17: 193-200.

## 2019

* Shiquan Sun, Jiaqiang Zhu, Ying Ma and Xiang Zhou (2019). Accuracy, robustness and scalability of dimensionality reduction methods for single-cell RNA-seq analysis. Genome Biology. 20: 269. 
* Yue Fan, Tauras P. Vilgalys, Shiquan Sun, Qinke Peng, Jenny Tung and Xiang Zhou (2019). IMAGE: High-powered detection of genetic effects on DNA methylation using integrated methylation QTL mapping and allele-specific analysis. Genome Biology. 20: 220.
* Ping Zeng, Ting Wang, Junnian Zheng and Xiang Zhou (2019). Causal association of type 2 diabetes on amyotrophic lateral sclerosis: New evidence from Mendelian randomization using GWAS summary statistics. BMC Medicine. 17: 225.
* Patricia M Schnepp, Mengjie Chen, Evan T Keller and Xiang Zhou (2019). SNV identification from single-cell RNA sequencing data. Human Molecular Genetics. 28: 3569–3583.
* Ping Zeng, Xinghao Yu and Xiang Zhou (2019). Birth weight is not causally associated with adult asthma: results from instrumental variable analyses. Scientific Reports. 9:7647.
* Ping Zeng and Xiang Zhou (2019). Causal association between birth weight and adult diseases: Evidence from a Mendelian randomization analysis. Frontier in Genetics. 10:618.
* Ping Zeng and Xiang Zhou (2019). Causal effects of blood lipids on amyotrophic lateral sclerosis: A Mendelian randomization study. Human Molecular Genetics. 28: 688-697.
* Shiquan Sun\*, Jiaqiang Zhu\*, Sahar Mozaffari, Carole Ober, Mengjie Chen and Xiang Zhou (2019). Heritability estimation and differential analysis of count data with generalized linear mixed models in genomic sequencing studies. Bioinformatics. 35: 487-496.

## 2018

* Mengjie Chen and Xiang Zhou (2018). VIPER: Variability-preserving imputation for accurate gene expression recovery in single cell RNA sequencing studies. Genome Biology. 19:196.
* Ping Zeng, Xingjie Hao and Xiang Zhou (2018). Pleiotropic mapping and annotation selection in genome-wide association studies with penalized Gaussian mixture models. Bioinformatics. 34: 2797-2807. 
* Lorin Crawford#, Kris C. Wood, Xiang Zhou# and Sayan Mukherjee# (2018). Bayesian approximate kernel regression with variable selection. Journal of the American Statistical Association. 113: 1710-1721.
* Xingjie Hao, Ping Zeng, Shujun Zhang and Xiang Zhou (2018). Identifying and exploiting trait-relevant tissues with multiple functional annotations in genome-wide association studies. PLOS Genetics. e1007186.

## 2017

* Mengjie Chen and Xiang Zhou (2017). Controlling for confounding effects in single cell RNA sequencing studies using both control and target genes. Scientific Reports. 7: 13587. 
* Jingjing Yang, Lars G. Fritsche, Xiang Zhou# and Goncalo Abecasis# (2017). A scalable Bayesian method for integrating functional information in genome-wide association studies. American Journal of Human Genetics. 101: 404-416.
* Ping Zeng and Xiang Zhou (2017). Non-parametric genetic prediction of complex traits with latent Dirichlet process regression models. Nature Communications. 8: 456.
* Lorin Crawford, Ping Zeng, Sayan Mukherjee and Xiang Zhou (2017). Detecting epistasis with the marginal epistasis test in genetic mapping studies of quantitative traits. PLOS Genetics. e1006869. 
* Xiang Zhou (2017). A unified framework for variance component estimation with summary statistics in genome-wide association studies. Annals of Applied Statistics. 11(4): 2027-2051.
* Shiquan Sun, Michelle Hood, Laura Scott, Qinke Peng, Sayan Mukherjee, Jenny Tung and Xiang Zhou (2017). Differential expression analysis for RNAseq using Poisson mixed models. Nucleic Acids Research. 45(11): e106.

## 2016

* Noah Snyder-Mackler, William H Majoros, Michael L Yuan, Amanda O Shaver, Jacob B Gordon, Gisela H Kopp, Stephen A Schlebusch, Jeffrey D Wall, Susan C Alberts, Sayan Mukherjee, Xiang Zhou# and Jenny Tung# (2016). Efficient genome-wide sequencing and low-coverage pedigree analysis from non-invasively collected samples. Genetics. 203: 699-714.

## 2015

* Amanda J. Lea, Jenny Tung# and Xiang Zhou# (2015). A flexible, efficient binomial mixed model for identifying differential DNA methylation in bisulfite sequencing data. PLOS Genetics. 11: e1005650.
* Jenny Tung\*, Xiang Zhou\*, Susan C Alberts, Matthew Stephens and Yoav Gilad (2015). The genetic architecture of gene expression levels in wild baboons. eLife. 4: e04729.

## 2014

* Xiang Zhou\*, Carolyn Cain\*, Marsha Myrthil, Noah Lewellen, Katelyn Michelini, Emily Davenport, Matthew Stephens, Jonathan Pritchard and Yoav Gilad (2014). Epigenetic modifications are associated with inter-species gene expression variation in primates. Genome Biology. 15(12): 547. 
* Xiang Zhou# and Matthew Stephens# (2014). Efficient multivariate linear mixed model algorithms for genome-wide association studies. Nature Methods. 11(4): 407–409. 

## 2010-2013

* Xiang Zhou#, Peter Carbonetto and Matthew Stephens# (2013). Polygenic modeling with Bayesian sparse linear mixed models. PLOS Genetics. 9(2): e1003264.
* Xiang Zhou# and Matthew Stephens# (2012). Genome-wide efficient mixed-model analysis for association studies. Nature Genetics. 44: 821–824.
* Xiang Zhou, Jun Takatoh, Fan Wang (2011). The mammalian class 3 PI3K (PIK3C3) is required for early embryogenesis and cell proliferation. PLOS One. 6(1): e16358.

## 2003 - 2010

* Xiang Zhou, Jerome Reiter (2010). A note on Bayesian inference after multiple imputation. The American Statistician. 64(2): 159-163. 
* Xiang Zhou, Fan Wang (2010). Effects of neuronal PIK3C3/Vps34 deletion on autophagy and beyond. Autophagy. 6(6): 798-799. 
* Xiang Zhou, Liangli Wang, Hiroshi Hasegawa, Priyanka Amin, Bao-Xia Han, Shinjiro Kaneko, Youwen He, Fan Wang (2010). Deletion of PIK3C3/Vps34 in sensory neurons causes rapid neurodegeneration by disrupting the endosomal but not the autophagic pathway. Proc Natl Acad Sci U.S.A. 107(20): 9424-9429.
* Xiang Zhou\*, J. Ramesh Babu\*, Susana da Silva\*, Qing Shu, Isabella A. Graef, Tim Oliver, Toshifumi Tomoda, Tomomi Tani, Marie W. Wooten, and Fan Wang (2007). Unc-51-like kinase 1/2-mediated endocytic processes regulate filopodia extension and branching of sensory axons. Proc Natl Acad Sci U.S.A. 104(14): 5842-5847.
* Xiang Zhou, Jin-Ju Li, Long-Chuan Yu (2003). Plastic changes of calcitonin gene-related peptide in morphine tolerance: a behavioral and immunohistochemical study in rats. Journal of Neuroscience Research. 74: 622-629. 